src/tabletraits.jl

```diff
@@ -0,0 +1,72 @@
+using TableTraits
+
+TableTraits.isiterable(x::IndexedTable) = true
+TableTraits.isiterabletable(x::IndexedTable) = true
+
+function TableTraits.getiterator{S<:IndexedTable}(source::S)
+    return rows(source)
+end
+
+# Sink
+
+@generated function _fillIndexedTable{idx_indices,data_indices}(iter,idx_storage,data_storage,::Type{idx_indices},::Type{data_indices})
+    push_exprs = Expr(:block)
+    for (i,idx) in enumerate(map(i->i.parameters[1],idx_indices.parameters))
+        ex = :( push!(idx_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    for (i,idx) in enumerate(map(i->i.parameters[1],data_indices.parameters))
+        ex = :( push!(data_storage.columns[$i], row[$idx]) )
+        push!(push_exprs.args, ex)
+    end
+
+    quote
+        for row in iter
+            $push_exprs
+        end
+    end
+end
+
+function IndexedTable(x; idxcols::Union{Void,Vector{Symbol}}=nothing, datacols::Union{Void,Vector{Symbol}}=nothing)
```

Comment:
Oh, one more thing: we should also add a code path to this method that deals with an iterator where the element type is ``Pair{X,S}``. If it is just any ``Pair``, it would create an unnamed index and data column. If either ``X`` or ``X`` are a ``NamedTuple``, it would create named columns for the index and data columns. At that point the following would automatically work:
````julia
@from i in source begin
    @select {i.a, i.b} => {i.c,i.d}
    @collect IndexedTable
end
````
Not in this PR, but could be added later.